### DISCUSSION TITLE
Formating dates does not work when using globals

### DISCUSSION THREAD

**Discussion by SalahAdDin** [amannn/next-intl#869] [Q&A] [answered]
### Description

We are setting up some global formats for our website:
```ts title="i18n.ts"
const formats: Partial<Formats> = {
  dateTime: {
    short: {
      day: "numeric",
      month: "short",
      year: "numeric",
    },
  },
};

export default getRequestConfig(async ({ locale }) => {
  if (!isValidLocale(locale)) notFound();

  const messages = {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
    home: { ...(await import(`./locales/${locale}/home.json`)).default },
  };

  return {
    messages,
    formats,
  };
});
```

And we use it on our component:
```tsx title="component.tsx"
export const ContinueReadingArticleCard = ({
  category,
  publishedDate,
  title,
  thumbnail,
  slug,
}: ContinueReadingArticleCardProps) => {
  const format = useFormatter();

  return (
    <Link href={`/${category.slug}/${slug}`}>
      <article className={styles.continueReadingArticleCard}>
        <NextImage
          alt={thumbnail.alt}
          title={thumbnail.alt}
          className={styles.thumbnail}
          src={thumbnail.url}
          width={thumbnail.width || 1966}
          height={thumbnail.height || 1107}
          sizes="100vw"
        />
        <main>
          <p>
            <strong>{category.title}</strong>
            {format.dateTime(publishedDate,"short")}
          </p>
          <h6 className={styles.title}>{title}</h6>
        </main>
      </article>
    </Link>
  );
};
```

But it fails to format the date:
![image](https://github.com/amannn/next-intl/assets/5034215/71ed17ea-2251-4878-82cc-40562c5355a4)

### Mandatory reproduction URL

https://codesandbox.io/p/devbox/next-intl-bug-template-app-forked-v8pgz2

### Reproduction description

Steps to reproduce:
1. Setup your globals.
2. Run the app.
3. Verify it does not work.

### Expected behaviour

Now, if we use the format directly:
```tsx title="component.tsx"
export const ContinueReadingArticleCard = ({
  category,
  publishedDate,
  title,
  thumbnail,
  slug,
}: ContinueReadingArticleCardProps) => {
  const format = useFormatter();

  return (
    <Link href={`/${category.slug}/${slug}`}>
      <article className={styles.continueReadingArticleCard}>
        <NextImage
          alt={thumbnail.alt}
          title={thumbnail.alt}
          className={styles.thumbnail}
          src={thumbnail.url}
          width={thumbnail.width || 1966}
          height={thumbnail.height || 1107}
          sizes="100vw"
        />
        <main>
          <p>
            <strong>{category.title}</strong>
            {format.dateTime(publishedDate, {
              day: "numeric",
              month: "short",
              year: "numeric",
            })}
          </p>
          <h6 className={styles.title}>{title}</h6>
        </main>
      </article>
    </Link>
  );
};
```

It works:
![image](https://github.com/amannn/next-intl/assets/5034215/d0d21179-17ed-4030-9994-5d451f3cc930)

That's it what we expect to get from globals.

**Comment by amannn** [comment, marked as answer]
In the CodeSandbox you've provided, I can unfortunately not reproduce the issue:

![Screenshot 2024-02-16 at 21 41 55](https://github.com/amannn/next-intl/assets/4038316/ec754f59-6913-45ae-9a5c-c2c1f3803510)

Note that `formats` aren't automatically inherited by Client Components, maybe that's the issue you're seeing? In that case you can pass them explicitly to [`NextIntlClientProvider`](https://next-intl-docs.vercel.app/docs/usage/configuration#nextintlclientprovider).

**Comment by SalahAdDin** [comment]
@amannn Yeah, I saw it and I don't understand why it is happening.

This example is in a server component rendered in a story (storybook) that is wrapped in a `NextIntlClientProvider`:
![image](https://github.com/amannn/next-intl/assets/5034215/80c64b98-6f03-4d6c-945d-ae359f820cb7)
![image](https://github.com/amannn/next-intl/assets/5034215/d70d9c69-699d-4d66-910f-f4c2021b3b61)
![image](https://github.com/amannn/next-intl/assets/5034215/8c20e7e6-3d4a-4776-a9f8-cfe20d73d2fc)

**Comment by amannn** [comment]
You need to add `formats` on `NextIntlClientProvider`.

I'll move this to a discussion since it seems to be a usage question.

**Reply by SalahAdDin** [reply]
In the case of the storybook, right? It must be working normally in NextJS.

**Reply by amannn** [reply]
Yep, depending on if you're rendering a component as a Server Component (default in Next.js) or a Client Component (Storybook, `'use client'`, …), either [the config from `i18n.ts` or `NextIntlClientProvider`](https://next-intl-docs.vercel.app/docs/usage/configuration#client-server-components) will be applied.
